benchmark/reconstruction/download.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2023, ETH Zurich and UNC Chapel Hill.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#
+#     * Neither the name of ETH Zurich and UNC Chapel Hill nor the names of
+#       its contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import shutil
+import subprocess
+from pathlib import Path
+
+
+def download_eth3d(data_path: Path) -> None:
+    for filename, category in [
+        ("multi_view_training_dslr_undistorted.7z", "dslr"),
+        ("multi_view_test_dslr_undistorted.7z", "dslr"),
+        ("multi_view_training_rig_undistorted.7z", "rig"),
+        ("multi_view_test_rig_undistorted.7z", "rig"),
+    ]:
+        target_folder = data_path / category
+        target_folder.mkdir(parents=True, exist_ok=True)
+        subprocess.check_call(
+            ["wget", "-c", "https://www.eth3d.net/data/" + filename],
```

Comment:
Our C++ DownloadFile function won't work well for large files, because it doesn't stream the data in chunks to disk but instead keeps everything in memory. For the intended use case on the C++ case this is sufficient for now but it will fail for the files here, as some of them are multiple 10s of GB big. I'll annotate the DownloadFile() documentation to mention this limitation.